### PR TITLE
Deprecate Axes3D.w_{x,y,z}axis in favor of .{x,y,z}axis.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

See discussion at https://github.com/matplotlib/matplotlib/pull/11385#discussion_r193397411 for rationale.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 1]
```diff
@@ -192,19 +192,31 @@ def set_top_view(self):
...
+    def w_xaxis(self):
+        return self.xaxis
+
+    @property
+    @cbook.deprecated("3.0", alternative="yaxis")
+    def w_yaxis(self):
+        return self.yaxis
+
+    @property
+    @cbook.deprecated("3.0", alternative="zaxis")
```
Comment (WeatherGod) [inline lib/mpl_toolkits/mplot3d/axes3d.py]: We should give an explicit removal date. These attributes have been here for 10+ years, and I still see them used in submitted examples and questions. So the usual deprecation cycle may not be long enough for most people.

Comment (timhoffm) [reply lib/mpl_toolkits/mplot3d/axes3d.py]: +1 to take it slow. It's not hurting much. One could go for a pending deprecation warning first, and maybe target removal="4.0".

**Comment by anntzer** [comment]
By default, `cbook.warn_deprecated("3.0")` will emit a warning with a scheduled removal in 3.2.  I don't particularly like scheduling removal in 4.0 (that's way too vague), so I completely removed the scheduled deprecation instead.  (Is 3.3 enough? 3.4? 3.5? the question is a bit pointless IMO.)
I also don't see the point of PendingDeprecationsWarnings, tbh...
Or feel free to propose a better solution.

**Comment by timhoffm** [comment]
As I understand it, PendingDeprecationWarnings are helpful for third party packages using matplotlib. If they use `ax.w_zaxis` and we deprecate it right away, the change will be visible to end users and 3rd party devs at the same time. The end users will see the deprecation messages, but cannot do anything about it because it's not in their code.

On the other hand, a PendingDeprecationWarning is ignored by default. The end user will not see it. But the 3rd party package dev can enable these warnings in his tests. He then can recognize and fix a use of a to-be-deprecated element without the end user seeing a message about this ever.

That said, I seems we are always generating a `MatplotlibDeprecationWarning`, even when `warn_deprecated(..., pending=True) - other than indicated by the docstring. We should actually be issuing a `PendingDeprecationWarning` in that case.

**Comment by anntzer** [comment]
Changed the deprecation to a PendingDeprecationWarning with no scheduled removal yet.

File: lib/matplotlib/cbook/deprecation.py [hunk 2]
```diff
@@ -24,16 +24,16 @@ class MatplotlibDeprecationWarning(UserWarning):
 def _generate_deprecation_warning(
         since, message='', name='', alternative='', pending=False,
         obj_type='attribute', addendum='', *, removal=''):
-
```
Comment (anntzer) [inline lib/matplotlib/cbook/deprecation.py]: This patch doesn't change the logic, just clarifies that pending deprecations have no scheduled removal.

**Comment by tacaswell** [comment]
I think I see what happened here.  This got merged after we branched for 3.1, it did not get backported, to v3.1.x, but when we did the API notes consalidation it was against master branch in https://github.com/matplotlib/matplotlib/pull/13870 we _did_ pick up that change so we currently have this documented as deprecated at https://matplotlib.org/3.1.0/api/api_changes.html#axes3d but the code is not actually warning.

**Pull request merged**